### PR TITLE
DevOps: Deploy to production 

### LINKED ISSUE #270: DevOps: Promote OSMP to production instance
- Design the production deployment of the OSMP project. 
- Evaluate the next steps and internal and external requirements for the deployment. 
- Consult CHEO and PhenoTips for the next steps.

### PR DESCRIPTION
**Pull request by jennyziyi-xu**
- Github workflow files are created for deploying React, `server` and `keycloak` containers in production. 
- `docker-compose.prod.yaml` is created for deploying `server` and `keycloak` containers. 
- I created a branch `production` for code deployed in production. The rules for this branch can be found in Settings. In particular, commits can not be pushed directly to it, they can only be pushed through merge requests. Created an environment `OSMP_PROD` for all secrets used in production. Only the branch `production` can deploy to this environment and a member of “ccmbio/osmp” needs to approve the deployment for it to proceed.

- [x] @frewmack The "keycloak" sections are identical to those in the files for the staging instance. Please feel free to modify if necessary. 
- [ ] This PR is pending on domain DNS registration, MongoDB database and integration with Phenotips staging or production. 

Here is a list of secrets, feel free to add secrets to the environment `OSMP_PROD` when they are obtained. 

React:

- [x]  REACT_APP_GRAPHQL_URL (pending on domain name)
- [x]  REACT_APP_KEYCLOAK_AUTH_URL
- [x]  REACT_APP_KEYCLOAK_REALM
- [x]  REACT_APP_KEYCLOAK_CLIENT_ID
- [x] REACT_SOURCE_LIST
- [x] MC_HOST
- [x] MINIO_BUCKET

Server and Mongo:

- [x] G4RD_AUTH_METHOD
- [x]  G4RD_PASSWORD
- [x]  G4RD_URL
- [x]  G4RD_USERNAME
- [x]  G4RD_REALM
- [x]  G4RD_GRANT_TYPE
- [x]  G4RD_TOKEN_URL
- [x]  G4RD_CLIENT_ID
- [x]  G4RD_AUTH0_BASE_URL
- [x]  MONGO_CONNECTION_STRING (pending on account creation)

Keycloak

- [x]  KEYCLOAK_DB_ADDR
- [x]  KEYCLOAK_DB
- [x]  KEYCLOAK_DB_USER
- [x]  KEYCLOAK_DB_PASSWORD
- [x]  KEYCLOAK_REALM
- [x]  KEYCLOAK_AUTH_URL
- [x]  KEYCLOAK_CLIENT_ID

SSH Config

- [x]  DEPLOY_PRIVATE_KEY
- [x]  DEPLOY_HOST_KEY
- [x]  DEPLOY_SSH_HOST

### CODE REVIEW

File: docker-compose.prod.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      KEYCLOAK_USER:
+      KEYCLOAK_PASSWORD:
+      KEYCLOAK_REALM:
+      DB_VENDOR: mysql
+      DB_ADDR: "${KEYCLOAK_DB_ADDR}"
+      DB_DATABASE: "${KEYCLOAK_DB}"
+      DB_USER: "${KEYCLOAK_DB_USER}"
+      DB_PASSWORD: "${KEYCLOAK_DB_PASSWORD}"
+      PROXY_ADDRESS_FORWARDING: "true"
+      KEYCLOAK_FRONTEND_URL: "https://keycloak.ccmdev.ca/auth"
```
Comment (frewmack) [inline docker-compose.prod.yaml]:
```suggestion
      KEYCLOAK_FRONTEND_URL: "https://keycloak.osmp.genomics4rd.ca/auth"
```

**Review by frewmack — CHANGES_REQUESTED** [review]

**Comment by frewmack** [comment]
Added Keycloak secrets. Credentials for linking Keycloak to G4RD's Auth0 instance will have to be put in place after production goes up.

**Comment by themisteryang** [comment]
Added the `MONGO_CONNECTION_STRING` secret to the `OSMP_PROD` environment!

**Review by themisteryang — APPROVED** [review]
LGTM! 🚀

**Review by frewmack — APPROVED** [review]
Changes look good! I think we can merge this to develop, and then merge our first release from develop onto production once the proxy changes are pushed on the CHEO RI infrastructure repo.

**Comment by jennyziyi-xu** [comment]
Sounds good to me!

**Pull request merged**